runtime/dart_init.cc

```diff
@@ -216,12 +230,14 @@ Dart_Isolate ServiceIsolateCreateCallback(const char* script_uri,
       const bool disable_websocket_origin_check = false;
       const bool service_isolate_booted = DartServiceIsolate::Startup(
           ip, port, tonic::DartState::HandleLibraryTag,
-          IsRunningPrecompiledCode(), disable_websocket_origin_check, error);
+          IsRunningPrecompiledCode() || is_running_from_kernel,
```

Comment:
This is going to be confusing inside DartServiceIsolate::Startup method as the parameter there states 'running_precompiled' which won' be true in this case.

Maybe the parameter 'running_precompiled' should be changed to 'running_fromsource' ?

Reply:
Done